lib/Basics/process-utils.cpp

```diff
@@ -335,29 +355,32 @@ static void StartExternalProcess(ExternalProcess* external, bool usePipes,
                             &file_actions, &spawn_attrs, external->_arguments,
                             envs.data());
 
-  posix_spawnattr_destroy(&spawn_attrs);
-  posix_spawn_file_actions_destroy(&file_actions);
-
-  bool executableNotFound = false;
+  ScopeGuard cleanup([&]() noexcept {
+    posix_spawnattr_destroy(&spawn_attrs);
+    posix_spawn_file_actions_destroy(&file_actions);
+  });
```

Comment:
This scope guard should be active as soon as the `init` calls have succeeded, because `reserve` and `push_back` might throw a `std::bad_alloc` exception.

Reply:
Have moved it up.